### ISSUE TITLE
PODM 2.1.3 trouble allocating a node

### ISSUE THREAD

**Issue by geoffdillon** [intel/intelRSD#98]
I'm trying to test our integration with pod manager 2.1.3 and I'm having a problem composing a very simple node and I can't figure out where the failure is.  Here is my json request:

`{"Memory": [{"CapacityMiB": 2048}], "Processors": [{"TotalCores": 2}], "Name": "FirstNode", "Description": "FirstNode"} `

I keep getting the same response:
```
{
  "error" : {
    "code" : "Base.1.0.ResourcesStateMismatch",
    "message" : "Conflict during allocation",
    "@Message.ExtendedInfo" : [ {
      "Message" : "There are no computer systems available for this allocation request."
    }, {
      "Message" : "Available assets count after applying filters: [available: 5 -> status: 0 -> resource: 0 -> chassis: 0 -> processors: 0 -> memory: 0 -> local drives: 0 -> ethernet interfaces: 0 -> computer system attributes: 0]"
    } ]
  }
}
```

This is the response from https://10.3.0.1:8443/redfish/v1/Systems
```
{
  "@odata.context" : "/redfish/v1/$metadata#Systems",
  "@odata.id" : "/redfish/v1/Systems",
  "@odata.type" : "#ComputerSystemCollection.ComputerSystemCollection",
  "Name" : "Computer System Collection",
  "Description" : "Computer System Collection",
  "Members@odata.count" : 5,
  "Members" : [ {
    "@odata.id" : "/redfish/v1/Systems/1-s-rack1-block1-sled2-node1"
  }, {
    "@odata.id" : "/redfish/v1/Systems/1-s-rack1-block2-sled5-node1"
  }, {
    "@odata.id" : "/redfish/v1/Systems/1-s-rack1-block2-sled1-node1"
  }, {
    "@odata.id" : "/redfish/v1/Systems/1-s-rack1-block1-sled3-node1"
  }, {
    "@odata.id" : "/redfish/v1/Systems/1-s-rack1-block1-sled1-node1"
  } ]
}
```

and here is a list of what the systems report for their resources
```
Id  |State     |Location           |MAC               |Asset#    |Serial# |Pwr |MFG   |CPUs |Cores|HDDs |HDD GiB|Memory GiB|
--------------------------------------------------------------------------------------------------------------------
1   |StandbyOffline|Rack1,Block1,Sled2 |50:9a:4c:97:d2:de |          |CNWS30087M00H1|Off |Dell  |2    |32   |5    |20006  |29.802336 |
2   |StandbyOffline|Rack1,Block2,Sled5 |50:9a:4c:97:d2:29 |          |CNWS30087M00DG|Off |Dell  |2    |32   |1    |447    |29.802336 |
3   |StandbyOffline|Rack1,Block2,Sled1 |50:9a:4c:97:d2:44 |          |CNWS30087M00AC|Off |Dell  |2    |32   |1    |447    |29.802336 |
4   |Enabled   |Rack1,Block1,Sled3 |50:9a:4c:97:d2:e4 |          |CNWS30087M009H|On  |Dell  |2    |32   |3    |1378   |29.802336 |
5   |Enabled   |Rack1,Block1,Sled1 |80:18:44:7a:17:be |          |CN70163724009L|On  |Dell  |2    |36   |0    |0      |119.209344|
```

**Comment by geoffdillon** [comment]
I have tried specifying more details such as which Chassis /rack / block / sled to use but it hasn't made any difference.

**Comment by RobertCMa** [comment]
Hi @geoffdillon ,

I was called in to assist this thread. Is this issue still active? If yes, some questions will need your clarification.

It can see two systems are enabled and 3 systems are StandbyOffline. What's the Status/Health in these systems?  Is there curl log, which can help elaborate the situation of these systems?

In the meanwhile, is there Intel contact supporting this RSD 2.1.3 integration but not the latest RSD 2.3.2 ?

**Comment by geoffdillon** [comment]
Hi Robert,

Yes this is still active although the test system was reallocated for other purposes since I wasn't able to make any progress.  Yes we do have an Intel contact working with us on CTS certification on 2.1.3.  I wanted to open this issue in order to get a more technical answer to some of the issues we seem to have with the Pod Manager itself.  I am not sure why we chose a much older version but I assume it was because of a prior commitment.

I believe the other systems were simply powered off.   I assumed that the two online systems would have enough resources to service the request.  

I will have to negotiate some time with the test system and do some work to get it back into intel RSD setup before I can do more debugging.

**Comment by RobertCMa** [comment]
Hi @geoffdillon,

Understand that it may take some time to setup the environment. As there is Intel contact working with you on the 2.1.3 project with commitment, that's suggested that you can send emails with this issue to the contact and include me in that loop. In case the project information is not proper to show up in the Github and we can work more efficiently with the same topic.

In the meanwhile, you can refer to the chapter 6.1.9.1 "Detailed process of selecting and allocating  a Computer System for a Composed Node" in the spec of RSD 2.1 PODM user guide and check the Status/Health in these systems first.

**Comment by geoffdillon** [comment]
Hi @RobertCMa,

Fortunately we have resolved the issue.  The problem was that the physical compute systems were reporting a Critical Health state due to a loose power cord.   Unfortunately I couldn't see that state reflected in the response from PODM, just a general failure with no explanation.   I have updted our scripts that send commands to pod-manager to check the health of the systems before trying to take action on them.  Thanks for your help...
